app/MMB/src/list/FlightList.js

```diff
@@ -0,0 +1,274 @@
+                idx(current, _ => _.node.multicity),
+              ),
+            },
+          ];
+        default:
+          return acc;
+      }
+    }, []);
+  };
+
+  oneWayToFlightDetail = (flight: ?OneWayType) => {
+    return {
+      departureDate: idx(flight, _ => _.trip.departure.time),
+      departureCity: idx(flight, _ => _.trip.departure.airport.city.name),
+      status: idx(flight, _ => _.status),
+      bookingId: idx(flight, _ => _.databaseId),
+      finalDestination: idx(flight, _ => _.trip.arrival.airport.city.name),
+    };
+  };
+
+  returnToFlightDetail = (flight: ?ReturnType) => ({
+    departureDate: idx(flight, _ => _.outbound.departure.time),
+    departureCity: idx(flight, _ => _.outbound.departure.airport.city.name),
+    status: idx(flight, _ => _.status),
+    bookingId: idx(flight, _ => _.databaseId),
+    finalDestination: idx(flight, _ => _.outbound.arrival.airport.city.name),
+  });
+
+  multicityToFlightDetail = (flight: ?MulticityType) => ({
+    departureDate: idx(flight, _ => _.start.time),
+    departureCity: idx(flight, _ => _.start.airport.city.name),
+    status: idx(flight, _ => _.status),
+    bookingId: idx(flight, _ => _.databaseId),
+    finalDestination: idx(flight, _ => _.end.airport.city.name),
+  });
+
+  render = () => (
+    <ScrollView contentContainerStyle={styles.container}>
+      <Text style={styles.upcomingText}>
+        <Translation id="mmb.my_bookings.future_trips" />
+      </Text>
+      {this.getFlightDetails().map(item => (
+        <View key={item.bookingId} style={styles.itemContainer}>
+          <CityImage
+            imageUrl={item.imageUrl || ''}
+            bookingId={item.bookingId}
+            type={item.type || ''}
+            departureCity={item.departureCity || ''}
+            arrivalCity={item.finalDestination || ''}
+            date={item.departureDate}
+            status={item.status || ''}
+            passengerCount={item.passengerCount}
+          />
+        </View>
+      ))}
+    </ScrollView>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    padding: 10,
+  },
+  itemContainer: {
+    marginBottom: 16,
+  },
+  upcomingText: {
+    marginTop: 10,
+    marginBottom: 12,
+    color: Color.textLight,
+  },
+});
+
+export default createRefetchContainer(
+  FlightList,
+  graphql`
+    fragment FlightList on RootQuery {
+      allBookings {
+        edges {
+          node {
+            destinationImageUrl(dimensions: _375x165)
+            type
+            passengerCount
+            oneWay {
+              databaseId
+              status
+              trip {
+                departure {
+                  time
+                  airport {
+                    city {
+                      name
+                    }
+                  }
+                }
+                arrival {
+                  airport {
+                    city {
+                      name
+                    }
+                  }
+                }
```

Comment:
This repetition leads to the custom fragment which leads to a separate component. Try to **compose** it a little bit differently, please. 🙂